bundles/org.openhab.binding.matter/src/main/java/org/openhab/binding/matter/internal/controller/devices/converter/SwitchConverter.java

```diff
@@ -0,0 +1,132 @@
+ *
+ * @author Dan Cunningham - Initial contribution
+ */
+@NonNullByDefault
+public class SwitchConverter extends GenericConverter<SwitchCluster> {
+    private Gson gson = new Gson();
+
+    public SwitchConverter(SwitchCluster cluster, MatterBaseThingHandler handler, int endpointNumber,
+            String labelPrefix) {
+        super(cluster, handler, endpointNumber, labelPrefix);
+    }
+
+    @Override
+    public Map<Channel, @Nullable StateDescription> createChannels(ChannelGroupUID thingUID) {
+        final Map<Channel, @Nullable StateDescription> map = new java.util.HashMap<>();
+        Set<ChannelTypeUID> triggerChannelTypes = new HashSet<>();
+        // See cluster specification table 1.13.4. Switch Features
+        if (initializingCluster.featureMap.latchingSwitch) {
+            triggerChannelTypes.add(CHANNEL_SWITCH_SWITCHLATECHED);
+        }
+        if (initializingCluster.featureMap.momentarySwitch) {
+            triggerChannelTypes.add(CHANNEL_SWITCH_INITIALPRESS);
+        }
+        if (initializingCluster.featureMap.momentarySwitchRelease) {
+            triggerChannelTypes.add(CHANNEL_SWITCH_SHORTRELEASE);
+        }
+        if (initializingCluster.featureMap.momentarySwitchLongPress) {
+            triggerChannelTypes.add(CHANNEL_SWITCH_LONGPRESS);
+            triggerChannelTypes.add(CHANNEL_SWITCH_LONGRELEASE);
+        }
+        if (initializingCluster.featureMap.momentarySwitchMultiPress) {
+            triggerChannelTypes.add(CHANNEL_SWITCH_MULTIPRESSCOMPLETE);
+            triggerChannelTypes.add(CHANNEL_SWITCH_MULTIPRESSONGOING);
+        }
+        triggerChannelTypes.forEach(type -> map.put(ChannelBuilder.create(new ChannelUID(thingUID, type.getId()), null)
+                .withType(type).withKind(ChannelKind.TRIGGER).build(), null));
+
+        Channel channel = ChannelBuilder
+                .create(new ChannelUID(thingUID, CHANNEL_ID_SWITCH_SWITCH), CoreItemFactory.NUMBER)
+                .withType(CHANNEL_SWITCH_SWITCH).build();
+
+        List<StateOption> options = new ArrayList<>();
+        for (int i = 0; i < initializingCluster.numberOfPositions; i++) {
+            options.add(new StateOption(String.valueOf(i), "Position " + i));
+        }
+
+        StateDescription stateDescriptionMode = StateDescriptionFragmentBuilder.create().withPattern("%d")
+                .withOptions(options).build().toStateDescription();
+
+        map.put(channel, stateDescriptionMode);
+
+        return map;
+    }
+
+    @Override
+    public void onEvent(AttributeChangedMessage message) {
+        Integer numberValue = message.value instanceof Number number ? number.intValue() : 0;
+        switch (message.path.attributeName) {
+            case SwitchCluster.ATTRIBUTE_CURRENT_POSITION:
+                initializingCluster.currentPosition = numberValue;
+                updateState(CHANNEL_ID_SWITCH_SWITCH, new DecimalType(numberValue));
```

Comment:
Should you update channel state only if message.value is a Number or is it OK to set channel state to 0 in other case ?
I ask because you did it differently in this converter compared to other converters.